### ISSUE TITLE
Null ref: T2.Creature.Slay – ShowingTitleScreen

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#742]
Auto report, 0.2.1... creature trying to be slayed while title screen is showing.

Feedback ID: 20201213-8XM43

```
1:13:56, Frame 516325, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.Creature.Slay (UnityEngine.Vector2 fromPosition, System.Boolean forceZaerieExplosionNow)
T2.QueuedCreatureSlayer.Update ()

GameStates: ShowingTitleScreen
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.2.3.

Believe it was a case of: creature is set to die, user pauses, then returns to title screen, and something is null within `Creature.SlayOrFallOver` since no dungeon would have been active.

`Creature.SlayOrFallOver` is a massive method. Added further null checks within. If issue persists, will need to break down the slay method into smaller methods.

**Comment by ChrisNZL** [comment]
Related auto report from 0.2.2a:
- Feedback ID: 20210115-MTF7Y

```
2:48:31, Frame 596570, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.NetIDDatabase.ImportRoomBlob (T2.NetworkedRoomStructBlob blob)
T2.Creature.SlayOrFallOver (UnityEngine.Vector2 fromPosition, System.Boolean forceZaerieExplosionNow)
T2.QueuedCreatureSlayer.Update ()

GameStates: ShowingTitleScreen
OnlineGameInfo: Device is offline.
No previous server nor online game info.
```

`QueuedCreatureSlayer.Update` – the fix applied should handle this now. If player is no longer in a dungeon, this method (which resumes once game's `Time.timeScale` returns to `1f`) will destroy the queued creatures immediately, rather than trying to assume the player is romping through a dungeon.
